### ISSUE TITLE
TSQL: BULK INSERT causes parsing error

### ISSUE THREAD

**Issue by borchero** [sqlfluff/sqlfluff#4380]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The [BULK INSERT](https://learn.microsoft.com/en-us/sql/t-sql/statements/bulk-insert-transact-sql?view=sql-server-ver16) query below caused a parsing error.

### Expected Behaviour

No parsing error.

### Observed Behaviour

```
== [test.sql] FAIL                                                                                                                         
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "BULK
                       | INSERT mytable\nFROM 'data.csv';"
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```sql
BULK INSERT mytable
FROM 'data.csv';
```

### Dialect

`tsql`

### Version

`sqlfluff, version 1.4.5`

### Configuration

None.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
